druid/src/widget/disable_if.rs

```diff
@@ -0,0 +1,50 @@
+use crate::{
+    BoxConstraints, Data, Env, Event, EventCtx, LayoutCtx, LifeCycle, LifeCycleCtx, PaintCtx,
+    Point, Size, UpdateCtx, Widget, WidgetPod,
+};
+
+/// A widget wrapper which disables the inner widget if the provided closure return true.
+pub struct DisabledIf<T, W> {
+    inner: WidgetPod<T, W>,
```

Comment:
Because this widget does not modify the position of the child, and does not need to be addressed independently of the child (e.g. doesn't need its own `WidgetId`) we can skip using `WidgetPod` here and just embed the child directly; see `IdentityWrapper` for another example of this.

Reply:
If we don't use a `WidgetPod`, the inner widget could overwrite the disabled state. In reality this will probably never happen, i'm not sure if we still should try to prevent this?

Reply:
That's a good point, I think it's an argument to keep the `WidgetPod`.

Reply:
Om :)